File: src/libs/ThirdParty.CSSuppliers/DOTW/IDOTWSettings.cs
namespace ThirdParty.CSSuppliers.DOTW
{

    public interface IDOTWSettings
    {
        string CompanyCode(IThirdPartyAttributeSearch tpAttributeSearch);
        string CustomerCountryCode(IThirdPartyAttributeSearch tpAttributeSearch);
        string CustomerNationality(IThirdPartyAttributeSearch tpAttributeSearch);
        int DefaultCurrencyID(IThirdPartyAttributeSearch tpAttributeSearch);
        bool ExcludeDOTWThirdParties(IThirdPartyAttributeSearch tpAttributeSearch);
        string Password(IThirdPartyAttributeSearch tpAttributeSearch);
        bool AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch);
        string RequestCurrencyID(IThirdPartyAttributeSearch tpAttributeSearch);
        bool UseGZip(IThirdPartyAttributeSearch tpAttributeSearch);
        int OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch, bool isMandatory);
        bool SendTradeReference(IThirdPartyAttributeSearch tpAttributeSearch);
        string ServerURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string ThreadedSearch(IThirdPartyAttributeSearch tpAttributeSearch);
        bool UseMinimumSellingPrice(IThirdPartyAttributeSearch tpAttributeSearch);
        string Username(IThirdPartyAttributeSearch tpAttributeSearch);
        string Version(IThirdPartyAttributeSearch tpAttributeSearch);
    }

}

File: src/libs/ThirdParty.VBSuppliers/Properties/DOTW/IDOTWSettings.vb
Imports ThirdParty.Abstractions

Public Interface IDOTWSettings
        ReadOnly Property CompanyCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property CustomerCountryCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property CustomerNationality(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property DefaultCurrencyID(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
        ReadOnly Property ExcludeDOTWThirdParties(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
        ReadOnly Property Password(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property AllowCancellations(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
        ReadOnly Property RequestCurrencyID(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property UseGZip(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
        ReadOnly Property OffsetCancellationDays(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As Integer
        ReadOnly Property SendTradeReference(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
        ReadOnly Property ServerURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property ThreadedSearch(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property UseMinimumSellingPrice(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
        ReadOnly Property Username(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property Version(tpAttributeSearch As IThirdPartyAttributeSearch) As String
End Interface
